### PR TITLE
Sync: Fix how we enqueue term_relationships on full sync 🏝

### PR DESCRIPTION
**Pull request by lezama**
Fixes how we iterate over al `term_relationships` items.

#### Testing instructions:
* Find a site with a lot of `term_relationships` 
* Disable sync on your site. (you can use the debugger)
* Add some `term_relationships`
* Start a full sync that includes `term_relationships` (from wpcom: `wp jetpack-sync start --blog-id=144249210 --object-type=term_relationships`)
* Make sure the new `term_relationships` get sync.
* Test `test_full_sync_sends_all_term_relationships` should fail on `master` and pass on this branch

### CODE REVIEW

**Comment by lezama** [comment]
I am adding tests

**Comment by lezama** [comment]
new test is failing 😅

File: packages/sync/src/Settings.php [hunk 1]
```diff
@@ -29,27 +29,28 @@ class Settings {
...
+		'network_disable'               => true,
+		'render_filtered_content'       => true,
+		'post_meta_whitelist'           => true,
+		'comment_meta_whitelist'        => true,
+		'max_enqueue_full_sync'         => true,
+		'max_queue_size_full_sync'      => true,
+		'sync_via_cron'                 => true,
+		'cron_sync_time_limit'          => true,
+		'known_importers'               => true,
+		'term_relationships_batch_size' => true,
```
Comment (lezama) [inline packages/sync/src/Settings.php]: maybe `term_relationships_item_size`?

File: packages/sync/src/modules/Term_Relationships.php [hunk 2]
```diff
@@ -74,52 +92,55 @@ public function init_before_send() {
...
-		$action_name           = 'jetpack_full_sync_term_relationships';
 
-		// Count down from max_id to min_id so we get term relationships for the newest posts and terms first.
-		// phpcs:ignore WordPress.CodeAnalysis.AssignmentInCondition.FoundInWhileCondition, WordPress.DB.PreparedSQL.InterpolatedNotPrepared
-		while ( $ids = $wpdb->get_results( "SELECT object_id, term_taxonomy_id FROM $wpdb->term_relationships WHERE object_id <= {$previous_interval_end['object_id']} AND term_taxonomy_id < {$previous_interval_end['term_taxonomy_id']} ORDER BY object_id DESC, term_taxonomy_id DESC LIMIT {$items_per_page}", ARRAY_A ) ) {
+		while ( $limit > 0 ) {
+			// Count down from max_id to min_id so we get term relationships for the newest posts and terms first.
+			$objects = $wpdb->get_results( $wpdb->prepare( "SELECT object_id, term_taxonomy_id FROM $wpdb->term_relationships WHERE ( object_id = %d AND term_taxonomy_id < %d ) OR ( object_id < %d ) ORDER BY object_id DESC, term_taxonomy_id DESC LIMIT %d", $last_object_enqueued['object_id'], $last_object_enqueued['term_taxonomy_id'], $last_object_enqueued['object_id'], $limit ), ARRAY_A );
```
Comment (enejb) [inline packages/sync/src/modules/Term_Relationships.php]: Lets write this nicer, its hard to read.

Comment (lezama) [reply packages/sync/src/modules/Term_Relationships.php]: what do you propose?

Comment (lezama) [reply packages/sync/src/modules/Term_Relationships.php]: I added a comment with a query example

**Review by enejb — APPROVED** [review]
Thanks for all the changes! 

I really like the new approach and the code is much cleaner! 

Thanks for also adding and fixing the tests here!

File: packages/sync/src/modules/Term_Relationships.php [hunk 3]
```diff
@@ -7,14 +7,32 @@
...
+	const QUERY_LIMIT = 1000;
+
+	/**
+	 * Max mysql integer
+	 *
+	 * @access public
+	 *
+	 * @const int
+	 */
+	const MAX_INT = 999999999;
```
Comment (griffbrad) [inline packages/sync/src/modules/Term_Relationships.php]: @lezama Sorry for the drive-by comment.  Happened to stumble across this PR and noticed this `MAX_INT` definition.  Should this be 2147483647 -- the max value for an unsigned INT in MySQL?  (https://dev.mysql.com/doc/refman/8.0/en/integer-types.html

Comment (lezama) [reply packages/sync/src/modules/Term_Relationships.php]: it is a good comment! I don't remember how I decided on that, but I do remember checking that MySQL documentation for it, I think somehow I thought it would be too  magical, I will change it to `2147483647` with a good code comment 👍

Comment (lezama) [reply packages/sync/src/modules/Term_Relationships.php]: https://github.com/Automattic/jetpack/pull/14042
